lib/registration.pm

```diff
@@ -93,6 +92,17 @@ sub fill_in_registration_data {
         }
     }
 
+    # Soft-failure for module preselection
+    if (sle_version_at_least('15') && check_var('DISTRI', 'sle')) {
+        assert_screen("modules-preselected-" . get_required_var('SLE_PRODUCT'));
```

Comment:
Can this be check_screen since it is for soft failure? and that would not block the upgrade test :)
https://openqa.suse.de/tests/1167547#step/patch_before_migration/25

Reply:
Oh, I have missed needles for text mode. In general this soft-failure is applicable for upgrade as well, as we should pre-select modules. I'll create a needle for that.

Reply:
Actually after checking that I see that modules are shown for SLE 12, but VERSION variable states 15. Shouldn't it be 12sp3 then? I'll add check that it's not PATCH. But it's tricky as then during installation. I'll create PR for that.

Reply:
Please, review https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3572